google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -98,24 +101,71 @@ def rows
               @metadata ||= grpc.metadata
               @stats ||= grpc.stats
 
-              if chunked_value
-                grpc.values.unshift merge(chunked_value, grpc.values.shift)
-                chunked_value = nil
-              end
-              to_iterate = values + grpc.values
-              chunked_value = to_iterate.pop if grpc.chunked_value
+              buffered_responses << grpc
+
+              if (grpc.resume_token && grpc.resume_token != "") ||
+                buffered_responses.size >= buffer_upper_bound
+                # This can set the resume_token to nil
```

Comment:
In this specific case I felt a longer, unfactored method was better than implementing the wrong abstraction. It allows for easier changes.

We can't simply break the look up into smaller methods, since it all needs to execute within the same method context because of the Enumerator. It can probably be better, but I'd rather have it working and unfactored now than fully factored sometime in the future.